components/Editor.js

```diff
@@ -62,8 +60,21 @@ class Editor extends React.Component {
 
   async componentDidMount() {
     const { asPath = '' } = this.props.router
-    const { query } = url.parse(asPath, true)
-    const initialState = getQueryStringState(query)
+    let { state: initialState, gistUri } = getRouteState(asPath)
+
+    if (this.context.gist && gistUri) {
+      try {
+        const gistState = await this.context.gist.get(gistUri)
+
+        initialState = {
```

Comment:
If a gist is found, it should use the gist state as is, not mix it with the localStorage state.

Reply:
Basically, if it doesn't find a config from the gist, then it can look in localStorage. Everything elsse stays the exact same,